src/ResourceManager/Automation/Commands.Automation/Common/PowershellJsonConverter.cs

```diff
@@ -71,7 +71,7 @@ public static PSObject Deserialize(string json)
         /// <returns></returns>
         private static Collection<PSObject> InvokeScript(string scriptName, Hashtable parameters)
         {
-            using (var powerShell = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace))
+            using (var powerShell = System.Management.Automation.PowerShell.Create(RunspaceMode.NewRunspace))
```

Comment:
where is this used, and what are the implications of using a new runspace versus the current runspace?

Reply:
Please check https://github.com/Azure/azure-powershell/pull/5750/files#diff-bae4b09865ecee44819a2743713e4b1bR1929

I am calling the PowerShellJsonConverter.Serialize(inputObject)  in new Thread to handle timeout.

When we use the CurrentRunspace i saw the below exception and that is why changed to NewRunspace

**System.InvalidOperationException: A Windows PowerShell object cannot be created that uses the current runspace because there is no current runspace available.  The current runspace might be starting, such as when it is created with an Initial Session State.**